release/release_tests.yaml

```diff
@@ -4404,7 +4404,7 @@
 ###############
 
 - name: inference
-  group: core-dataset-tests
+  group: Data tests
```

Comment:
It'd be better to have a consistent naming scheme, like "data-tests", or as we will have more tests coming in, we can group them into themes like "data-api-benchmarks" etc.

Reply:
+1 to `data-tests` rather than `Data tests`.

Reply:
hmm may i ask why we want to differ from other libraries? RLlib tests, Train tests, Serve tests, etc.

Reply:
Or "data_tests". These are structured fields which could be used to query the dashboard so it's more conventional to name in this scheme, as the rest of those fields here.

Reply:
Sorry my question is why want to different from other libraries naming?

examples:
https://github.com/ray-project/ray/blob/master/release/release_tests.yaml#L126
https://github.com/ray-project/ray/blob/master/release/release_tests.yaml#L2632
https://github.com/ray-project/ray/blob/master/release/release_tests.yaml#L2658

Our top-level comment is kind of recommending `group: Example group` as well - https://github.com/ray-project/ray/blob/master/release/release_tests.yaml#L8 .

Anything I am missing here?

Reply:
I think both @jianoaix and I are pushing for hyphen/underscore separators since this field will be used in dashboard queries, since it's less ambiguous on how different systems will tokenize the query. E.g. will "Data tests" translate to a single EQUALS/CONTAINS query on a "Data tests" token, or an OR query for "Data" and "tests" tokens?

When determining the separator to use in a manifest field string, I usually go with:
1. If only read by humans, go with more readable space separator.
2. If only used programmatically, go with less ambiguous hyphen/underscore separator.
3. If used by both, go with (2).

Reply:
Yeah, for example this dashboard (we use filter queries to narrow down tests):
<img width="949" alt="Screen Shot 2023-01-06 at 10 42 03 AM" src="https://user-images.githubusercontent.com/99709935/211077973-c1792b09-7697-42ca-a96b-28ebb9347443.png">

We may look at this in two issues:
- There is inconsistency in naming the group: this pre-exists this PR and we may need to fix it (we probably all agree on this) but doesn't need to be in this PR
- Then which naming format we want to unify on: I'd prefer hyphen/underscore like Clark.

Reply:
ah got the point from your guys now. I didn't think about the dashboard use case, then this makes sense to me. Updated to `data-tests`. Thanks.